src/write.c

```diff
@@ -2206,8 +2528,9 @@ static avifResult avifRWStreamWriteProperties(avifItemPropertyDedup * const dedu
         avifEncoderItem * item = &encoder->data->items.item[itemIndex];
         const avifBool isGrid = (item->gridCols > 0);
         const avifBool isToneMappedImage = !memcmp(item->type, "tmap", 4);
+        const avifBool isSampleTransformImage = !memcmp(item->type, "sato", 4);
         memset(&item->ipma, 0, sizeof(item->ipma));
-        if (!item->codec && !isGrid && !isToneMappedImage) {
+        if (!item->codec && !isGrid && !isToneMappedImage && !isSampleTransformImage) {
```

Comment:
Optional: If would be good to avoid the `isToneMappedImage` and `isSampleTransformImage` code completely when the respective experimental feature flag is disabled. Here is one way to do that:

```
        const avifBool isGrid = (item->gridCols > 0);
        // Whether there is ipma to write for this item
        avifBool hasIpmaToWrite = item->codec || isGrid;
#if defined(AVIF_ENABLE_EXPERIMENTAL_GAIN_MAP)
        const avifBool isToneMappedImage = !memcmp(item->type, "tmap", 4);
        if (isToneMappedImage) {
            hasIpmaToWrite = AVIF_TRUE;
        }
#endif
#if defined(AVIF_ENABLE_EXPERIMENTAL_SAMPLE_TRANSFORM)
        const avifBool isSampleTransformImage = !memcmp(item->type, "sato", 4);
        if (isSampleTransformImage) {
            hasIpmaToWrite = AVIF_TRUE;
        }
#endif
        memset(&item->ipma, 0, sizeof(item->ipma));
        if (!hasIpmaToWrite) {
            continue;
        }
```

Reply:
Done.